tests/unit/apps/test_auxiliary.py

```diff
@@ -30,91 +31,113 @@
     UpgradeStep,
 )
 from cou.utils import app_utils
+from cou.utils.juju_utils import COUMachine, COUUnit
 from cou.utils.openstack import OpenStackRelease
 from tests.unit.apps.utils import add_steps
+from tests.unit.utils import assert_steps
 
 
-def test_auxiliary_app(status, config, model, apps_machines):
-    # version 3.8 on rabbitmq can be from ussuri to yoga. In that case it will be set as yoga.
-    expected_units = [
-        ApplicationUnit(
-            name="rabbitmq-server/0",
-            os_version=OpenStackRelease("yoga"),
-            workload_version="3.8",
-            machine=apps_machines["rmq"]["0/lxd/19"],
-        )
-    ]
+def test_auxiliary_app(model):
+    """Test auxiliary application.
 
+    The version 3.8 on rabbitmq can be from ussuri to yoga. In that case it will be
+    set as yoga.
+    """
+    machines = {"0": MagicMock(spec_set=COUMachine)}
     app = RabbitMQServer(
-        "rabbitmq-server",
-        status["rabbitmq_server"],
-        config["auxiliary_ussuri"],
-        model,
-        "rabbitmq-server",
-        apps_machines["rmq"],
+        name="rabbitmq-server",
```

Comment:
I'm against creating a application instance every time to test something. It's very repetitive and does not make part of the test itself, so I think it should be moved somehow to conftest.

I know that the current unit tests are not ideal, but this can also add pain for the future. If a new field is addes/removed this would result into a lot of broken unit tests and the changes will be spread instead of changing on a single place